swarm/src/connection.rs

```diff
@@ -42,13 +42,36 @@ use libp2p_core::connection::ConnectedPoint;
 use libp2p_core::multiaddr::Multiaddr;
 use libp2p_core::muxing::{StreamMuxerBox, StreamMuxerEvent, StreamMuxerExt, SubstreamBox};
 use libp2p_core::upgrade::{InboundUpgradeApply, OutboundUpgradeApply};
-use libp2p_core::PeerId;
 use libp2p_core::{upgrade, UpgradeError};
+use libp2p_core::{Endpoint, PeerId};
 use std::future::Future;
 use std::task::Waker;
 use std::time::Duration;
 use std::{fmt, io, mem, pin::Pin, task::Context, task::Poll};
 
+/// Connection identifier.
+#[derive(Debug, Copy, Clone, Hash, PartialEq, Eq, PartialOrd, Ord)]
+pub struct ConnectionId(usize);
+
+impl ConnectionId {
+    /// Creates a `ConnectionId` from a non-negative integer.
```

Comment:
```suggestion
    /// Creates a [`ConnectionId`] from a non-negative integer.
```

Reply:
This will go away in https://github.com/libp2p/rust-libp2p/pull/3327 so I am not gonna bother :)